spec/carin_for_blue_button/carin_search_test_spec.rb

```diff
@@ -0,0 +1,315 @@
+# frozen_string_literal: true
+
+RSpec.describe CarinForBlueButtonTestKit::CarinSearchTest do
+  let(:eob_json_string) do
+    File.read(File.join(__dir__, '..', 'fixtures', 'c4bb_eob_inpatient_example.json'))
+  end
+
+  let(:suite) { Inferno::Repositories::TestSuites.new.find('c4bb_v200') }
+  let(:session_data_repo) { Inferno::Repositories::SessionData.new }
+  let(:test_session) { repo_create(:test_session, test_suite_id: suite.id) }
+  let(:url) { 'http://example.com/fhir' }
+  let(:error_outcome) { FHIR::OperationOutcome.new(issue: [{ severity: 'error' }]) }
+
+  def run(runnable, inputs = {})
+    test_run_params = { test_session_id: test_session.id }.merge(runnable.reference_hash)
+    test_run = Inferno::Repositories::TestRuns.new.create(test_run_params)
+    inputs.each do |name, value|
+      session_data_repo.save(
+        test_session_id: test_session.id,
+        name:,
+        value:,
+        type: runnable.config.input_type(name)
+      )
+    end
+    Inferno::TestRunner.new(test_session:, test_run:).run(runnable)
+  end
+
+  def setup_mock_test(search_test, eob_resource)
+    allow_any_instance_of(search_test).to receive(:scratch_resources).and_return(
+      {
+        all: [eob_resource]
+      }
+    )
+  end
+
+  describe 'carin search requiring service-date' do
+    let(:service_date_search_test) do
```

Comment:
This works. If you don't have any changes from the regular CarinSearchTest, you can use that directly here. Check the reference_test_spec in US Core. Benefit of doing that is this spec test uses the same metadata as the search_test.

Reply:
CarinSearchTest, is a module, I refactored to use the search param test class instead as it is done in US Core: CarinForBlueButtonTestKit::CARIN4BBV200::ExplanationOfBenefitServiceDateSearchTest, CarinForBlueButtonTestKit::CARIN4BBV200::ExplanationOfBenefitServiceStartDateSearchTest, etc.